### PR TITLE
Ruby - stop using an obsolete method of Net::HTTP

### PR DESCRIPTION
**Pull request by masakazutakewaka**
## Description
`Net::HTTP::Proxy` is an obsolete method, so replaced it with `Net::HTTP.new` as the doc says -> https://docs.ruby-lang.org/en/2.7.0/Net/HTTP.html#method-c-Proxy

### Motivation and Context
To keep the code up to date.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist
- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rb/lib/selenium/server.rb [hunk 1]
```diff
@@ -120,16 +120,16 @@ def latest
         end
       end
 
-      def net_http
+      def net_http_start(address, &block)
```
Comment (masakazutakewaka) [inline rb/lib/selenium/server.rb:123]: Should I write about this in the CHANGE.log as this method is exposed to public?

Comment (p0deje) [reply rb/lib/selenium/server.rb:123]: It might worth mentioning, but frankly, I don't think this was ever intended to be a public API.

**Comment by masakazutakewaka** [comment]
Hi @p0deje, it's green now.

**Comment by p0deje** [comment]
Thank you!

**Pull request merged**